Boundary documentation: modified Boundary's overview page (#2264)
hashicorp/boundary: website/content/docs/what-is-boundary.mdx

# What is Boundary?

HashiCorp Boundary is a tool for managing identity-based access for modern, dynamic infrastructure. Boundary’s workflow layers security controls and integrations on multiple levels monitoring and managing user access through:

- Tightly scoped identity-based permissions
- "Just-in-time" network and credential access for sessions via [HashiCorp Vault](https://vaultproject.io/)
- Single sign-on to target services and applications via external identity providers
- Access-as-code to automate the configuration of user permissions
- Automated discovery of target systems
- Session monitoring and management for access created via Boundary.

Boundary's foundation is based on the following important concepts.

- **Zero Trust Security:** Zero-Trust is an identity-based access model where the user access is continuously authenticated. Access is only authorized when the established rules and policies tied to the user’s identity are verified.

> - **Consistent Workflow for Access:** Once the user is verified and granted access, Boundary securetly connects the user securely to their infrastructure regardless of cloud platform, target environment, or identity provider. This foundation provides continuous user authentication and authorization workflows within ephemeral sessions, which administrators can monitor and manage securely.

Reviewer: ```suggestion
- **Consistent Workflow for Access:** Once the user is verified and granted access, Boundary securely connects the user securely to their infrastructure regardless of cloud platform, target environment, or identity provider. This foundation provides continuous user authentication and authorization workflows within ephemeral sessions, which administrators can monitor and manage securely.
```

- **Extensibility with the Ecosystem:** Modern organizations often require a multilayered access matrix constructed of identity providers, policy engines, secrets management tools, target types, and cloud providers that integrate and allow users to reside within access workflows requiring vendor lock-in. Boundary does not require vendor lock-in and supports the user's vendor-of-choice.

![Boundary workflow](/img/boundary.png)

## How does Boundary work?

Boundary provides secure access to hosts and critical systems without distributing and managing credentidals, configuring firewalls, or exposing the organization's private network. Traditionally, for users to access their resources, it's required that organizations establish and maintain SSH bastion hosts and VPNs. The illustration below displays Boundary's core workflow.

> ![Boundary core workflow](/img/boundary-core-workflow.png)

Reviewer: Image seems blurry in the preview

Author: I'll see about fixing that image to make it clearer.

Author: I played around with the image size and font. It won't be as clear as the image above it (different tools used) but hopefully it's readable and a tad bit better. Eventually, we may need the assistance of the marketing team to fine-tune some of the images. Long term, a new writer should be able to get in there and fix some of them or at least work with the marketing folks to help fine-tune.

### Filtering

Parts of Boundary support filters for various purposes. For a description
of the filter syntax, see the [filtering](/docs/concepts/filtering) page. See
the docs pages for the individual resources or capabilities where filters are
supported for the specific inputs and examples with those inputs.

-> **Tip**: Learn more about Boundary [use cases](/docs/use-cases).

## What is HCP Boundary?

Boundary offers two types of deployment options. A first option is an *OSS self-managed* deployment solution as discussed above. A self-managed approach enables organizations to proxy all session data through their own network while still providing the convenience of a managed service. A second option is an *HCP-managed* deployment solution where both the control plan and worker nodes are managed by HashiCorp. With this managed solution, an option of private workers is offered. **HCP Boundary** is a fully-managed, cloud-based workflow that enables secure connections to remote hosts and critical systems across cloud and on-premise environments. Refer to the [HCP Boundary](https://cloud.hashicorp.com/docs/boundary
) documentation to learn more.

> > **Hands On:** Try the [Create a Boundary Instance on HCP](https://learn.hashicorp.com/tutorials/boundary/hcp-getting-started-create?in=boundary/hcp-getting-started) tutorial to deploy an HCP Boundary instance.

Reviewer 2: ```suggestion
-> **Hands On:** Try the [Create a Boundary Instance on HCP](https://learn.hashicorp.com/tutorials/boundary/hcp-getting-started-create?in=boundary/hcp-getting-started) tutorial to deploy an HCP Boundary instance.
```

granted via [roles](/docs/concepts/domain-model/roles) that provide
authorization to create sessions against these targets.

### Filtering

Some parts of Boundary support filters for various purposes. For a description
of the filter syntax, see the [filtering](/docs/concepts/filtering) page. See
the docs pages for the individual resources or capabilities where filters are
supported for the specific inputs and examples with those inputs.

-> **Tip**: Learn more about Boundary [use cases](/docs/use-cases).

## What is HCP Boundary?

Boundary offers two types of deployment options. The first option is the *self-managed* approach, which we have discussed on this page. The self-managed approach enables organizations to proxy all session data through their own network while still providing the convenience of a managed service. **HCP Boundary**, on the other hand,  is a *fully-managed*, cloud-based workflow that enables secure connections to remote hosts and critical systems across cloud and on-premise environments. In the fully-managed deployment model, both the control plane and worker nodes are managed by HashiCorp. Refer to the [HCP Boundary](https://cloud.hashicorp.com/docs/boundary

> ) documentation to learn more.

Reviewer 3: I'd like a second opinion on the language in this section, since there are really three different deployment models being described here?

Self-managed: there is Boundary OSS, which is truly "self-managed".
HCP with self-managed workers: HCP Boundary hosts the control plane, while the organization self-manages their own workers.
Fully-managed: HCP hosts both the control plane and worker nodes.

On the Concepts page, we state that:

> There are several options for deploying Boundary:
> 
> 1. HCP Boundary as a fully-managed service
> 2. Boundary OSS for self-managed deployments
> 3. Dev mode for testing and learning purposes

We should make these statements consistent, and probably distinguish that HCP Boundary supports fully-managed and self-proxied deployment options, so folks don't assume Boundary OSS is the only option there.

@PPacent and @xingluw might want to weigh in on this bit.

Reviewer: I would prefer it be stated in 2 options: 
- OSS self managed
- HCP managed solution
Where the managed solution has an option of private workers
The dev mode mentioned above would be part of the OSS self managed

Reviewer 3: @jefferai mentioned to me that we shouldn't call dev mode a "deployment" option, either... but I see that's what's stated under the [concepts overview page](https://www.boundaryproject.io/docs/concepts) too

Author: I tweaked the content a bit based on xingluw's feedback.

> **Hands On:** Try the [Create a Boundary Instance on HCP](https://learn.hashicorp.com/tutorials/boundary/hcp-getting-started-create?in=boundary/hcp-getting-started) tutorial on HashiCorp Learn to deploy a Boundary instance.

## Tutorial

Refer to HashicCorp Learn for [Boundary](https://learn.hashicorp.com/boundary) to gain access to Boundary's series of featured tutorials to learn how to set up, configure, and administer Boundary.

## Community

We welcome questions, suggestions, and contributions from the community.

- Ask questions in [HashiCorp Discuss](https://discuss.hashicorp.com/c/boundary/50).

> - Read our [contributing guide](https://github.com/hashicorp/tutorials/blob/main/CONTRIBUTING.md).
> - [Submit an issue](https://support.hashicorp.com/hc/en-us/requests/new) for bugs and feature requests.

Reviewer 2: ```suggestion
- Read our [contributing guide](https://github.com/hashicorp/boundary/blob/main/CONTRIBUTING.md).
- [Submit an issue](https://github.com/hashicorp/boundary/issues) for bugs and feature requests.
```
